tests/test_interface.py

```diff
@@ -184,3 +186,22 @@ def test_dir_whitelist_basic(fast_motor):
     set_engineering_mode(True)
     eng_dir = dir(fast_motor)
     assert len(eng_dir) > len(user_dir)
+
+
+@pytest.mark.parametrize('cls', conftest.all_device_classes)
+def test_per_class_whitelist(cls):
+    if BaseInterface not in cls.mro():
+        ignore_list = {'.areadetector.', }
+        fully_qualified_name = f'{cls.__module__}.{cls.__name__}'
+        if any(name in fully_qualified_name for name in ignore_list):
+            pytest.skip()
+        pytest.skip(f'{cls} does not inherit from the interface')
```

Comment:
Probably doesn't matter but I kinda feel like the ignore_list check should be outside/before the BaseInterface if branch.

Reply:
Good call - refactored as such.